### DISCUSSION TITLE
Can we run specific jobs in all servers that are available, and few work on optimistic locking

### DISCUSSION THREAD

**Discussion by ereshzealous** [jobrunr/jobrunr#826] [Q&A]
Hi 
I have a question, We have few recurring jobs that were configured. We want to run few jobs in all the servers. Few using Jobrunr optimistic locking(running on node/instance of servers). 
Is there a way to achieve this through a config parameter

Thanks

**Comment by rdehuyss** [comment]
Hi @ereshzealous , this is not possible in the OSS version but it is possible in the Pro version using [`serverTags`](https://www.jobrunr.io/en/documentation/pro/server-tags/).

I know a Pro customer who does this with a little bit of custom code where it works great (15 lines of code). There, the job is duplicated for each `BackgroundJobServer`. 

But I'm afraid JobRunr (Pro) will not implement this in the near future out-of-the-box. The reason for this is that `BackgroundJobServers` can come and go (e.g. scale up, scale down, crash) and I currently cannot guarantee that when such a job is created, it will be processed by all servers (again as they can come and go...).

**Comment by ap-sree** [comment]
I tried few things in (application running in multiple nodes).


Initializing two instance of Background sever. 

- 1st instance initialized with DB storage provider 
-  2nd instance initialized with in memory storage provider. 

Any job requires to run only in on node is scheduled using DB based job scheduler and any job which requires to run on all node with in-memory based job scheduler.

**Reply by rdehuyss** [reply]
To be honest, I don't understand how the in-memory job storage provider can do jobs on different nodes. 

In-memory means for me current node only...

**Reply by ap-sree** [reply]
Yes. This works when same application is deployed in multi-node config. In-memory instance of jobrunr will be running in each application node.

1st application node (1st Jobrunr instance with in-memory based and 2nd Jobrunr instance with DB based)
....
2nd application node (1st Jobrunr instance with in-memory based and 2nd Jobrunr instance with DB based)


JOB - A - run in all node - This will be scheduled using 1st instance jobrunr
JOB - B - run in only one node - This will be scheduled using 2nd instance jobrunr

JOB-A will be running in respective nodes. JOB-B will be running in only one node
